app/guided.py

```diff
@@ -0,0 +1,424 @@
+
+import collections
+import copy
+import heapq
+import itertools
+import re
+
+import networkx as nx
+
+from app.objects.c_ability import Ability
+from app.objects.c_agent import Agent
+from app.objects.secondclass.c_goal import Goal
+
+
+NONLIMITED_FACT_REGEX = r'#{(.*?)}'
```

Comment:
These are similar to regular expressions used in base planning service:
https://github.com/mitre/caldera/blob/master/app/utility/base_planning_svc.py#L17
https://github.com/mitre/caldera/blob/master/app/utility/base_planning_svc.py#L28

It might be helpful to add comments to explain what each regex does, see above code for examples.

I feel "NONLIMITED_FACT_REGEX" is a slight misnomer since it matches both limited and nonlimited facts.